geniza/corpus/views.py

```diff
@@ -308,24 +315,15 @@ def get_context_data(self, **kwargs):
         document = self.get_object()
 
         # get unique set of canvases on this document from images and transcriptions
-        images = (
-            document.iiif_images()
-        )  # order override should be respected by iiif_images()
+        # order override should be respected by iiif_images()
+        images = document.iiif_images()
+
         for ed in document.digital_editions().all():
             for canvas_uri in ed.content_html.keys():
-                if canvas_uri not in [i["canvas"] for i in images]:
-                    # append placeholder image for each canvas not in iiif_images
-                    images.append(
-                        {
-                            "image": {
-                                "info": static("img/ui/all/all/image-unavailable.png"),
-                            },
-                            "label": "",
-                            "shelfmark": "",
-                            "canvas": canvas_uri,
-                            "excluded": False,
-                        }
-                    )
+                if canvas_uri and canvas_uri not in images:
```

Comment:
can `canvas_uri` be `None` here?

Reply:
It should not (and maybe will not) in normal operation, but I did run into it in my testing. It might have been from an outdated annotation with the canvas set incorrectly. Should I investigate further to track down why it happened? Or maybe remove this check and then investigate if we encounter an error here in QA/production?

Reply:
outdated canvas sounds plausible... I like your last suggestion, remove the check but see if we have any problems with fresh data in qa